examples/cfq/syntax_based/grammar.py

```diff
@@ -168,8 +168,34 @@ def __init__(self, grammar_str: str):
     first_rule = sub_rules[0]
     self.grammar_entry = first_rule[0]
 
+  def collect_syntax_tokens(self):
+    syntax_tokens = set()
```

Comment:
Since you are converting to list anyway, i would use a list to begin with.

Reply:
If I did that I needed to make sure that an item is not already in the list before adding it. I can change to that if it's necessary.

Reply:
ohh right, good point!